include/univalue.h

```diff
@@ -6,6 +6,20 @@
 #ifndef __UNIVALUE_H__
 #define __UNIVALUE_H__
 
+// Legacy code that uses UniValue typically makes countless copies of UniValue
+// objects. These copies are often unnecessary and costly. To help with
+// converting legacy code into more efficient std::move friendly code, enable
+// NO_UNIVALUE_COPY_OPERATIONS. This will disable all UniValue methods that
+// cause copying of elements, causing compile errors. Each of these compile
+// errors can then be fixed with e.g. std::move(), or if a copy is still
+// necessary by explicitly calling the .copy() method. Unfortunately it cannot
+// detect all cases, as a copy constructor is still necessary for collections.
+//
+// You can disable copy operations here globally, or in individual files by
+// setting it before the first univalue.h include.
+//
+//#define NO_UNIVALUE_COPY_OPERATIONS
```

Comment:
All the checks check for the inverse, so why not call this `WITH_UNIVAL...` and remove the `!` below?

Reply:
I thought that this way it is  easy to just `#define NO_UNIVALUE_COPY_OPERATIONS` in any file that wants to do the conversion. Doing the inverse is more difficult... But thinking about it, I could change the macro to something like this in `univalue.h`:

```cpp
#ifndef WITH_UNIVALUE_COPY_OPERATIONS
#define WITH_UNIVALUE_COPY_OPERATIONS 1
#endif
```

Then anyone who wants to do the conversion has to do this:
```cpp
#define WITH_UNIVALUE_COPY_OPERATIONS 0
```

Then all the checks in univalue.h would be just `#if WITH_UNIVALUE_COPY_OPERATIONS`

Reply:
done in the rebase to  ced6a36